src/apps/api/tests/test_competitions.py

```diff
@@ -0,0 +1,63 @@
+import json
+
+from django.urls import reverse
+from rest_framework.test import APITestCase
+
+from competitions.models import CompetitionParticipant
+from factories import UserFactory, CompetitionFactory, CompetitionParticipantFactory, PhaseFactory, LeaderboardFactory, \
+    ColumnFactory
+
+
+class CompetitionParticipantTests(APITestCase):
+    def setUp(self):
+        self.creator = UserFactory(username='creator', password='creator')
+        self.other_user = UserFactory(username='other_user', password='other')
+        self.comp = CompetitionFactory(created_by=self.creator)
+        PhaseFactory(competition=self.comp)
+        self.leaderboard = LeaderboardFactory(competition=self.comp)
+        ColumnFactory(leaderboard=self.leaderboard)
+
+    def _prepare_competition_data(self, url):
+        resp = self.client.get(url)
+        data = resp.data
+        data.pop('id')
+
+        # We don't want to post back the logo url, since it's expecting JSON data with
+        # the base64 of the logo in it
+        data["logo"] = None
+
+        # Just get the key from the task and pass that instead of the object
+        data["phases"][0]["tasks"] = [data["phases"][0]["tasks"][0]["key"]]
+        return data
+
+    # TODO: Do we have competition permissions tests?
+    # def test_cant_edit_someone_elses_competition?
+
+    def test_adding_organizer_accepts_creates_accepted_participant(self):
+        self.client.login(username='creator', password='creator')
+        url = reverse('competition-detail', kwargs={"pk": self.comp.pk})
+
+        # Get comp data to work with
+        data = self._prepare_competition_data(url)
+
+        data["collaborators"] = [self.other_user.pk]
+        resp = self.client.put(url, data=json.dumps(data), content_type="application/json")
+        assert resp.status_code == 200
+        assert CompetitionParticipant.objects.filter(user=self.other_user, competition=self.comp, status=CompetitionParticipant.APPROVED).count() == 1
```

Comment:
Break this line up maybe?
```suggestion
        assert CompetitionParticipant.objects.filter(
            user=self.other_user, 
            competition=self.comp,
            status=CompetitionParticipant.APPROVED
        ).count() == 1
```

Reply:
✔️